KataVista/utils/KataCommand.php

```diff
@@ -0,0 +1,126 @@
+
+        $proceed = 'Yes';
+        $wait = 'Not yet';
+        $exit = 'Stop';
+
+        do {
+            if (!isset($answer) || ($answer === $proceed)) {
+                $newDescriptionLetter = $this->pickNewDescription($io, ...$descriptionLetters);
+                if ($newDescriptionLetter === null) {
+                    return 1;
+                }
+
+                $descriptionLetters[] = $newDescriptionLetter;
+            }
+
+            if ($loopMode) {
+                $answer = $io->choice('Proceed to next iteration?', [$proceed, $wait, $exit], $proceed);
+            }
+        } while ($loopMode && $answer !== $exit);
+
+        $io->note('For the next iteration code, proceed with the command below:');
+        $io->text('./console.php vista:describe '.implode($descriptionLetters));
+
+        return 0;
+    }
+
+    private function pickNewDescription(SymfonyStyle $io, string ...$descriptionLetters): ?string
+    {
+        if (empty($descriptionLetters)) {
+            $io->success('Welcome to the "vista" kata.');
+            $io->text('Your goal is to update the Painter so it renders what is described below.');
+            $io->text('This command will act as a project manager / product owner, and will give you new instruction along the way.');
+            $io->text('There should already be a failing test. First, fix the test with `make all` for the following landmark, then use the command again as indicated in the end.');
+            $io->newLine(2);
+
+            $newDescription = new RuleMeadow();
+        } else {
+            $availableDescription = $this->vista->getVisible(...$descriptionLetters);
+            if (empty($availableDescription)) {
+                $io->error('No more landmark to describe available for the kata, you can open a PR to add some.');
+
+                return null;
+            }
+
+            $io->comment('Picking a new landmark from '.count($availableDescription).' available.');
+            $newDescription = $availableDescription[array_rand($availableDescription)];
+        }
+
+        $newDescriptionLetter = $newDescription->getLetter();
+        $io->text('<fg=green> ! New landmark spotted: '.$newDescription->getDescription().' ('.$newDescription->getLetter().')</>');
+
+        $descriptions = $this->vista->getDescribed($newDescriptionLetter, ...$descriptionLetters);
+
+        $existingTips = array_merge(
+            ...array_map(
+                fn (Description $description): array => $description->getLetter() === $newDescriptionLetter ? [] : $description->getTips(),
+                $descriptions
+            )
+        );
+        $newTips = array_filter(
+            $newDescription->getTips(),
+            fn (string $tip): bool => !in_array($tip, $existingTips)
+        );
+        if (!empty($newTips)) {
+            $io->section('Here some free tips!');
+            $io->listing($newTips);
+        }
```

Comment:
:thought_balloon: The system of tips seems great and will help others to organize the kata and have some dynamic guidance.
Bu I don't know why, but I didn't see the tips during our session.